components/experimental/src/dimension/provider/compact_count_ule.rs

```diff
@@ -65,12 +65,12 @@ impl AsULE for CompactCount {
     #[inline]
     fn from_unaligned(unaligned: Self::ULE) -> Self {
         let count = match unaligned.0 & 0b0000_0111 {
-            0 => Count::Zero,
-            1 => Count::One,
-            2 => Count::Two,
-            3 => Count::Few,
-            4 => Count::Many,
-            5 => Count::Other,
+            0 => PluralCategory::Zero,
```

Comment:
You can use `(unaligned.0 & 0b0000_0111) as PluralCategory`, as it's `repr(u8)` with stable discriminants

Reply:
```
error[E0605]: non-primitive cast: `u8` as `PluralCategory`
  --> components/experimental/src/dimension/provider/compact_count_ule.rs:67:22
   |
67 |         let count =  (unaligned.0 & 0b0000_0111) as PluralCategory;
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object


```

Reply:
https://github.com/unicode-org/icu4x/pull/5406